### PR TITLE
Improve Bash Completion

### PR DESCRIPTION
**Pull request by tisba**
This PR makes a breaking change the `forge completion` command, which generates completion scripts. It now accepts an argument to generate completion scripts for bash, zsh, fish or powershell.

Example:

```terminal
$ forge completion zsh > ./forge-zsh.sh
$ source ./forge-zsh.sh
```

This PR also adds dynamic completion for the following commands:

* root: `--ouput` is completed
* Test Case: Completion for `orga/test-case` references (or just `orga` where applicable)
* Test Case Launch:
  * `--region` and `--sizing` flags are completed
  * `test-case-file` and `nfr-check-file` have hints to filter for `*.js` and `*.yaml`
* Data Source commands do now complete the organisation

### CODE REVIEW

File: cmd/completion.go [hunk 1]
```diff
@@ -1,67 +1,141 @@
...
+# to enable it.  You can execute the following once:
+
+$ echo "autoload -U compinit; compinit" >> ~/.zshrc
+
+# To load completions for each session, execute once:
+$ yourprogram completion zsh > "${fpath[1]}/_yourprogram"
+
+# You will need to start a new shell for this setup to take effect.
+
+Fish:
```
Comment (zeisss) [inline cmd/completion.go:42]: the examples are missing powershell :)

**Comment by zeisss** [comment]
Awesome! ❤️

File: cmd/completion.go (hunk 1 above)
Comment (tisba) [reply cmd/completion.go:42]: That's copy & paste from the generated command by cobra 🤷‍♂️

Comment (zeisss) [reply cmd/completion.go:42]: ah ok, then good enough :)

**Review by zeisss — APPROVED** [review]

**Pull request merged**